app/services/catalog/create_request_body_from.rb

```diff
@@ -0,0 +1,30 @@
+module Catalog
+  class CreateRequestBodyFrom
+    attr_reader :result
+
+    def initialize(order, order_item, task)
+      @order = order
+      @order_item = order_item
+      @task = task
+    end
+
+    def process
+      svc_params_sanitized = Catalog::OrderItemSanitizedParameters.new(:order_item_id => @order_item.id).process.sanitized_parameters
```

Comment:
@eclarizio do we have to account for the order being cancelled while we wait for the inventories to be computed.

Reply:
I don't think we need to worry about it, if it's cancelled while we wait for inventories to be computed it's the same as if it gets cancelled while we wait for it to be approved. As long as the state of the order isn't "Completed", "Failed", or "Ordered", it's still cancelable.